.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+name: ci
+# Version 1.0.0
+# This workflow is triggered on push events to the repository
+# Please find readme.md for the usage of this workflow
+
+on:
+  push:
+
+jobs:
+  PREPARE-job:
+    runs-on: ubuntu-latest
+    outputs:
+      dockerfile_name: ${{ steps.parse_path.outputs.dockerfile_name }}
+      platform_tag: ${{ steps.parse_path.outputs.platform_tag }}
+      date: ${{ steps.date.outputs.date }}
+      proceed_valid: ${{ steps.set_proceed_flag.outputs.proceed_valid }}
+      devmode: ${{ steps.set_proceed_flag.outputs.devmode }}
+      directory: ${{ steps.parse_path.outputs.directory }}
+      files: ${{ steps.changed_files.outputs.files }}
+      platform: ${{ steps.parse_path.outputs.platform }}
+      runner_label: ${{ steps.set_runner_label.outputs.runner_label }}
+      noscan: ${{ steps.set_proceed_flag.outputs.noscan }}
+    
+    steps:
+      - name: Set default runner label
+        id: set_default_runner_label
+        run: |
+          echo "runner_label=ubuntu-latest" >> $GITHUB_OUTPUT
+
+      - name: Check required variables and secrets and set runner-default
+        id: check_vars_secrets
+        run: |
+          missing_vars=()
+          missing_secrets=()
+
+          # check Variables
+          if [ -z "${{ vars.DOCKERHUB_USERNAME }}" ]; then
+            missing_vars+=("DOCKERHUB_USERNAME")
+          fi
+
+          if [ -z "${{ vars.QUAYIO_USERNAME }}" ]; then
+            missing_vars+=("QUAYIO_USERNAME")
+          fi
+
+          if [ -z "${{ vars.ACACIA_BUCKETNAME }}" ]; then
+            missing_vars+=("ACACIA_BUCKETNAME")
+          fi
+
+          # check Secrets
+          if [ -z "${{ secrets.PAT_TOKEN }}" ]; then
+            missing_secrets+=("PAT_TOKEN")
+          fi
+
+          if [ -z "${{ secrets.DOCKERHUB_TOKEN }}" ]; then
```

Comment:
The logic here is that we have both Docker and quay but realistically, we need one or the other. Wonder if we can change the logic to have 
```bash
docker_endpoint = ()
if [ -z "${{ secrets.DOCKERHUB_TOKEN }}" ]; then
  docker_endpoint+= (DOCKERHUB_TOKEN)
fi
```
and combine that with 
```bash
if [ ${#missing_vars[@]} -ne 0 ] || [ ${#missing_secrets[@]} -ne 0 ] || [ ${#docker_endpoint[@]} -ge 1 ]; then
```

What do you think?

Reply:
The endpoint designing is really a good idea. But before the final stage, I strongly keep both of them:
1. I used both of them to verify some unique features from the repo registry.
2. `quay.io` for pawsey is not a team account. We have reached the private repo limits. Once we turn it into public, we can not reverse the behaviours. However, quay.io is our main image publishing station, and we need this site as the default choice. 
3. I really want to migrate `trivy` to [docker scout](https://docs.docker.com/scout/). However, it still needs some team plan. I'm testing this feature and need to use Dockerhub as well.
4. Keeping one of them is a good idea. At the final stage, we can turn off one of them and use the other as main repo.